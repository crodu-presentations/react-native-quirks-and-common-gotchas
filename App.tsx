import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const App = () => {
  let myText = 'hello world';
  myText = '';

  let myValue = 5;
  myValue = 0;

  return (
    <View testID="container" style={styles.container}>
      <Text style={styles.title}>Hello dev.js</Text>
      {!!myText && <Text style={styles.description}>{myText}</Text>}
      {!!myValue && <Text style={styles.description}>{myValue}</Text>}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'lightblue',
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 24,
    fontWeight: '600',
    color: 'black',
  },
  description: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: '#333',
  },
});

export default App;
