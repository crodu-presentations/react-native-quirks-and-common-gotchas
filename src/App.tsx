import React from 'react';
import { Text, View } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import { Typography } from './styles/fonts';

const App = () => {
  return (
    <View testID="container" style={styles.container}>
      <Text style={[styles.title, Typography.h1]}>Hello dev.js</Text>
      <Text style={[styles.title, Typography.h2]}>Czy to jest pogrubione?</Text>
      <Text style={[styles.body, Typography.body]}>Zażółć gęślą jaźń</Text>
      <Text style={[styles.body, Typography.bodyStrong]}>
        Pogrubione: Zażółć gęślą jaźń
      </Text>
    </View>
  );
};

const styles = EStyleSheet.create({
  container: {
    backgroundColor: 'lightblue',
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    '@media (min-height: 800) and (max-height: 1200)': {
      marginVertical: '25%',
    },
  },
  title: {
    textAlign: 'center',
    fontSize: 36,
    fontFamily: 'JosefinSans-Light',
    color: '#000',
  },
  bold: {
    fontWeight: 'bold',
  },
  body: {
    fontFamily: 'monospace',
    fontSize: 18,
    color: '#333',
  },
});

export default App;
