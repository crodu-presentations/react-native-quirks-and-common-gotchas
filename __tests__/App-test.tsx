import React from 'react';
import 'react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import App from '../src/App';

describe('App', () => {
  it('renders correctly', () => {
    renderer.create(<App />);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
