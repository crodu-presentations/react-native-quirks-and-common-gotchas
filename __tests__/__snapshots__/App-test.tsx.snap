// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightblue",
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
  testID="container"
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 24,
        "fontWeight": "600",
      }
    }
  >
    Hello dev.js
  </Text>
</View>
`;
